src/Native/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -1000,10 +1008,26 @@ REDHAWK_PALEXPORT Int32 PalGetProcessCpuCount()
 //Reads the entire contents of the file into the specified buffer, buff
 //returns the number of bytes read if the file is successfully read
 //returns 0 if the file is not found, size is greater than maxBytesToRead or the file couldn't be opened or read
-REDHAWK_PALEXPORT UInt32 PalReadFileContents(_In_z_ const TCHAR* fileName, _Out_writes_all_(cchBuff) char* buff, _In_ UInt32 cchBuff)
+REDHAWK_PALEXPORT UInt32 PalReadFileContents(_In_z_ const TCHAR* fileName, _Out_writes_all_(maxBytesToRead) char* buff, _In_ UInt32 maxBytesToRead)
 {
-    // UNIXTODO: Implement this function
-    return 0;
+    int fd = open(fileName, O_RDONLY);
+    if (fd < 0)
+    {
+        return 0;
+    }
+
+    UInt32 bytesRead = 0;
+
+    off_t fileLen = lseek(fd, 0, SEEK_END);
```

Comment:
I've misread the doc. I have thought that fstat for the case of a symlink returns the length of the path in the symlink, but that is only true for the lstat.